lib/api/CInferenceModelMetadata.cc

```diff
@@ -271,20 +276,30 @@ void CInferenceModelMetadata::hyperparameterImportance(
         case maths::analytics::boosted_tree_detail::E_SoftTreeDepthTolerance:
             hyperparameterName = CDataFrameTrainBoostedTreeRunner::SOFT_TREE_DEPTH_TOLERANCE;
             break;
+        // Not tuned directly.
+        case maths::analytics::boosted_tree_detail::E_MaximumNumberTrees:
+            hyperparameterName = CDataFrameTrainBoostedTreeRunner::MAX_TREES;
+            break;
+
+        // Incremental train hyperparameters.
         case maths::analytics::boosted_tree_detail::E_PredictionChangeCost:
+            if (m_Task != api_t::E_Update) {
+                continue;
+            }
```

Comment:
It seems surprising to me that this is in `hyperparameterImportance` if the task is not E_Update. Is this definitely needed?

Reply:
It turns out that this can be addressed in `CBoostedTreeHyperparameters` which should only return importances for relevant hyperparameters.